scripts/lint-pr-checks.sh

```
@@ -0,0 +1,246 @@
+# 1. Swift Version Consistency
+# =============================================================================
+echo "1. Checking Swift version consistency..."
+
+# Check Package.swift files for swift-tools-version (exclude third-party)
+for pkg in $(find "$PROJECT_DIR" -name "Package.swift" \
+    -not -path "*/DerivedData/*" \
+    -not -path "*/.build/*" \
+    -not -path "*/build/*" \
+    -not -path "*/llama.cpp/*" \
+    -not -path "*/.venv/*" \
+    -not -path "*/node_modules/*"); do
```

Shellcheck (0.11.0)

[warning] For loops over find output are fragile. Use find -exec or a while read loop.

(SC2044)